pkg/query-service/app/metrics/v4/helpers/sub_query.go

```diff
@@ -3,21 +3,54 @@ package helpers
 import (
 	"fmt"
 	"strings"
+	"time"
 
 	"go.signoz.io/signoz/pkg/query-service/constants"
 	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
 	"go.signoz.io/signoz/pkg/query-service/utils"
 )
 
+var (
+	sixHoursInMilliseconds = time.Hour.Milliseconds() * 6
+	oneDayInMilliseconds   = time.Hour.Milliseconds() * 24
+)
+
+// start and end are in milliseconds
+func which(start, end int64) (int64, int64, string) {
+	// If time range is less than 6 hours, we need to use the `time_series_v4` table
+	// else if time range is less than 1 day and greater than 6 hours, we need to use the `time_series_v4_6hrs` table
+	// else we need to use the `time_series_v4_1day` table
+	var tableName string
+	if end-start <= sixHoursInMilliseconds {
+		// adjust the start time to nearest 1 hour
+		start = start - (start % (time.Hour.Milliseconds() * 1))
+		tableName = constants.SIGNOZ_TIMESERIES_v4_LOCAL_TABLENAME
```

Comment:
Does this table have 1hr replacement logic?

Reply:
Yes, https://github.com/SigNoz/signoz-otel-collector/blob/0485ef8e0482f4fd7c188c8dd5ecf7b92280e683/exporter/clickhousemetricsexporter/clickhouse.go#L453

Reply:
right. thanks